staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -265,6 +265,15 @@ func (s *Serializer) unmarshal(into runtime.Object, data, originalData []byte) (
 		// to unmarshal directly into the object.
 		m := u.UnstructuredContent()
 		strictJSONErrs, err = kjson.UnmarshalStrict(data, &m)
+
+		// check for fatal errors first that won't get caught if unmarshaling
```

Comment:
we should address this separately from this PR. we don't want to decode twice... opened https://github.com/kubernetes/kubernetes/pull/109019 as an alternative